src/tqec/compile/tree/tree.py

```diff
@@ -143,15 +145,26 @@ def _annotate_detectors(
         k: int,
         manhattan_radius: int = 2,
         detector_database: DetectorDatabase | None = None,
+        database_path: Path = DEFAULT_DETECTOR_DATABASE_PATH,
+        only_use_database: bool = False,
         lookback: int = 2,
     ) -> None:
         if manhattan_radius <= 0:
             return
         self._root.walk(
             AnnotateDetectorsOnLayerNode(
-                k, manhattan_radius, detector_database, lookback
+                k, manhattan_radius, detector_database, only_use_database, lookback
             )
         )
+        # The database will have been updated inside the above function, and here at the end of the computation we save it
+        # to file:
+        if detector_database is not None:
+            detector_database.to_file(database_path)
+
+        # NB if database is None when passed in to .walk(AnnotateDetectorsOnLayerNode), then no database will be created
+        # in this function, the database will still be None afterwards and so nothing will be saved to file.
+        # Therefore, the 'do not use database' code path (which was initialised by setting the database to None) will
+        # produce nothing at the end, as desired.
```

Comment:
I would simplify that comment to:

```suggestion
        # Save the database to a file if it was not None.
        if detector_database is not None:
            detector_database.to_file(database_path)
```

Even that comment might not be needed.

Reply:
I agree with removing the second comment - it made sense to me in my head, but perhaps wouldn't to someone outside my head :) However, I think the first one is important, for others who like I did initially saw nothing happening in this level of the function and would then be confused by why an unchanged database would be being saved back to disc.

Reply:
thanks!